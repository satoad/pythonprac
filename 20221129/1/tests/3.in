class C(metaclass=dump):
	def __init__(self, val):
		self.val = val

	def add(self, other, another=None):
		self.val += other
		return self.val + other + (another or self.val)

c = C(10)
print(c.add(9))
print(c.add(10,another=10))
print(c.add(15))
print(c.add(20,another=30))
print(c.add(15))
print(c.add(20,another=30))
print(c.add(15))
print(c.add(5,another=12))
print(c.add(1))
print(c.add(2,another=3))
print(c.add(15))
print(c.add(10,another=5))