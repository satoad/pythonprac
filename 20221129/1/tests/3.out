__init__: (10,), {}
add: (9,), {}
47
add: (10,), {'another': 10}
49
add: (15,), {}
103
add: (20,), {'another': 30}
114
add: (15,), {}
173
add: (20,), {'another': 30}
149
add: (15,), {}
243
add: (5,), {'another': 12}
136
add: (1,), {}
241
add: (2,), {'another': 3}
127
add: (15,), {}
289
add: (10,), {'another': 5}
162
